libs/modelviewer/src/providers/selectionProvider.tsx

```diff
@@ -95,19 +95,22 @@ export const SelectionContextProvider = ({
     }
   }, [tagsOverlay]);
 
-  const { echoInstance } = useModelViewerContext();
-  const { modelMeta } = useModelContext();
-  const [currentNodes, setCurrentNodes] = useState<HierarchyNodeModel[]>([]);
+  const { selectionService } = useSelectionService();
 
-  const selectionService = useMemo(() => {
-    if (modelMeta && echoInstance) {
-      return new SelectionService(modelMeta, echoInstance);
-    }
-  }, [modelMeta, echoInstance]);
+  const [currentNodes, setCurrentNodes] = useState<HierarchyNodeModel[]>([]);
+  const [isTagFetching, setIsTagFetching] = useState<boolean>(true);
 
   useEffect(() => {
     if (tagList.length > 0 && selectionService) {
-      selectNodesByTagsOverlay(tagList).then((nodes) => nodes && setCurrentNodes(nodes));
+      selectNodesByTagsOverlay(tagList)
+        .then((nodes) => {
+          if (nodes) {
+            setCurrentNodes(nodes);
+          }
+        })
+        .finally(() => {
+          setIsTagFetching(false);
+        });
     }
   }, [tagList, selectionService]);
 
```

Comment:
Why is the cleanup in the useEffect below commented out?

Reply:
Fixed